protected/humhub/modules/content/search/SearchRecord.php

```diff
@@ -47,19 +47,36 @@ public function getImage(): string
     public function getTitle(): string
     {
         $title = RichTextToHtmlConverter::process($this->content->getContentDescription());
-        $title = str_replace(["\r", "\n"], ' ', strip_tags($title));
-
-        // Cut text to first word contained the searched keyword
-        $keywordIndex = strpos($title, $this->keyword);
-        if ($keywordIndex) {
-            do {
-                $keywordIndex--;
-                $titlePart = substr($title, $keywordIndex);
-            } while ($keywordIndex && $titlePart[0] !== ' ');
-            $title = '...' . trim($titlePart);
+        $title = preg_replace('/[\r\n\s]+/', ' ', strip_tags($title));
+        return $this->cutStringToKeyword($title);
+    }
+
+    /**
+     * Cut string to a word before first word contained the searched keyword
+     *
+     * @param string $string
+     * @param int $maxWordNumberBeforeKeyword
+     * @return string
+     */
+    protected function cutStringToKeyword(string $string, int $maxWordNumberBeforeKeyword = 1): string
+    {
+        $index = stripos($string, $this->keyword);
+
+        if ($index < 40) {
```

Comment:
Does that make sense? Often there will be no hit here either.

```suggestion
        if ($index === false || $index < 40) {
```

Reply:
ok, done https://github.com/humhub/humhub/pull/7195/commits/78cfa3d44d8d2eb34beae55fbc32859c5802db51#diff-c4cb0fcf397d1272c6dd36f78b88a309cf38639b6947ca4258ca8eccd8d25e93R65 to avoid some restriction/error in future php versions, but now it works well because `(false < 40) === true`.